base/array.jl

```diff
@@ -383,6 +383,9 @@ function setindex!{T}(A::Array{T}, X::Array{T}, c::Colon)
     return A
 end
 
+setindex!{T}(A::Array{T}, x::Number, c::Colon) = fill!(A, x)
```

Comment:
Oh, yeah, that's #3738 (unions with static type parameters are problematic).  Julia can either match the second argument to the `Number` in the union, or it can bind `T` to `Int` and match the typevar.  Julia chooses poorly.